### PR TITLE
Simplify security analyzer confirmation: replace two reject options with single 'Reject' option

### PR DESCRIPTION
**Pull request by malhotra5**
## Summary of PR

This PR simplifies the security analyzer confirmation interface in the OpenHands CLI by replacing two separate reject options with a single, more intuitive "Reject" option that prompts for an optional reason.

**Changes:**
- Replace "No, reject (w/o reason)" and "No, reject with reason" with single "Reject" option
- When "Reject" is selected, prompt with "Reason (ENTER to reject without reason): "
- Empty input results in rejection without reason, non-empty input passes as the reason
- Update all tests to match new behavior and option indices
- Add test for empty reason input handling

**User Experience Improvement:**
- Reduces cognitive load by eliminating redundant options
- Provides clearer workflow: select "Reject" → optionally provide reason
- Maintains all existing functionality while simplifying the interface

## Change Type

- [x] Refactor

## Checklist

- [x] I have read and reviewed the code and I understand what the code is doing.
- [x] I have tested the code to the best of my ability and ensured it works as expected.

## Release Notes

- [x] Include this change in the Release Notes.

Simplified security analyzer confirmation interface by replacing two reject options with a single "Reject" option that prompts for optional reason.

@malhotra5 can click here to [continue refining the PR](https://app.all-hands.dev/conversations/debb7d3461394b31bc78a674d858e033)

---

To run this PR locally, use the following command:

GUI with Docker:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:600730d-nikolaik   --name openhands-app-600730d   docker.all-hands.dev/all-hands-ai/openhands:600730d
```

CLI with uvx:
```
uvx --python 3.12 --from git+https://github.com/All-Hands-AI/OpenHands@simplify-security-confirmation-reject-options#subdirectory=openhands-cli openhands
```

### CODE REVIEW

File: openhands-cli/openhands_cli/user_actions/agent_action.py [hunk 1]
```diff
@@ -61,32 +60,18 @@ def ask_user_confirmation(
...
-        # Support both string return and (reason, cancelled) tuple for tests
-        cancelled = False
-        if isinstance(reason_result, tuple) and len(reason_result) >= 1:
-            reason = reason_result[0] or ''
-            cancelled = bool(reason_result[1]) if len(reason_result) > 1 else False
-        else:
-            reason = str(reason_result or '').strip()
-
-        if cancelled:
+            reason = cli_text_input('Reason (ENTER to reject without reason): ').strip()
```
Comment (enyst) [inline openhands-cli/openhands_cli/user_actions/agent_action.py]:
```suggestion
            reason = cli_text_input('Reason (and let OpenHands know why): ').strip()
```
It's always possible to enter an empty user message, if people want to, right? I don't know if we need to document that. I'd guess most of the time one doesn't want the LLM to do the thing, so they'd say something.

IIRC this is also Claude behavior, so I feel it's probably what users expect.
